File: Personal Development/advent-of-code-2023/day6/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
	timer "time"
)

func main() {
	st := timer.Now()
	file, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	lines := make([]string, 0)

	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}

	time := make([]string, 0, len(lines[0]))
	for _, c := range strings.Split(lines[0], " ") {
		if _, err := strconv.Atoi(c); err == nil {
			time = append(time, c)
		}
	}
	distance := make([]string, 0, len(lines[0]))
	for _, c := range strings.Split(lines[1], " ") {
		if _, err := strconv.Atoi(c); err == nil {
			distance = append(distance, c)
		}
	}

	time_, _ := strconv.Atoi(strings.Join(time, ""))
	distance_, _ := strconv.Atoi(strings.Join(distance, ""))
	tot := 0
	for j := 0; j < time_+1; j++ {
		if distance_ < (time_-j)*j {
			tot += 1
		}
	}
	fmt.Println(tot)
	fmt.Println(timer.Since(st))
}

File: Personal Development/advent-of-code-2023/day6/main.py
def solution():
    with open("input.txt", "r") as f:
        lines = f.readlines()
        lines = [line.strip() for line in lines]
    
    time = [int(i) for i in lines[0].split() if i.isdigit()]
    distance = [int(i) for i in lines[1].split() if i.isdigit()]

    total = 1
    for i in range(len(time)):
        tot = 0
        for j in range(1, time[i]+1):
            if distance[i] < (time[i]-j) * j:
                tot += 1
        
        print(tot, time[i], distance[i])
        total *= tot
    
    return total


if __name__ == "__main__":
    print(solution())